src/sections/reports.ts

```diff
@@ -64,7 +75,9 @@ interface GetReportRequestListParameters {
 const GetReportRequestListResult = Codec.interface({
   NextToken: optional(nextTokenCodec('GetReportRequestList')),
   HasNext: optional(boolean),
-  ReportRequestInfo: optional(ReportRequestInfo),
+  ReportRequestInfo: optional(oneOf([array(ReportRequestInfo), ReportRequestInfo, exactly('')])),
```

Comment:
@moltar It's this line, not sure why it's working, I tried digging into `ensureArray`'s source to see if there's something I'm missing but I couldn't understand what could be causing it to not decode properly. 

It's specifically for these two types of responses
https://github.com/ScaleLeap/amazon-mws-api-sdk/blob/feature/reports-api/test/unit/__fixtures__/reports_get_report_request_list.xml

https://github.com/ScaleLeap/amazon-mws-api-sdk/blob/feature/reports-api/test/unit/__fixtures__/reports_get_report_request_list_multiple.xml

When I use `ensureArray` on the XML files above it throws an error on the single ReportRequest fixture with   'Problem with value at index 0: Expected an object, but received undefined' and the error with multiple ReportRequests is 'Expected an object, but received an array'

Reply:
@gigobyte any idea on the above?

Reply:
`ensureArray` doesn't work either because the fixture is wrong or because the data is not structured like a regular array.

Let's compare different possible XML values and how they will be deserialized:

```javascript
// XML
<PaymentMethodDetails>
</PaymentMethodDetails>

// JSON
{ PaymentMethodDetails: "" }
```
```javascript
// XML
<PaymentMethodDetails>
    <PaymentMethodDetail>CreditCard</PaymentMethodDetail>
</PaymentMethodDetails>

// JSON
{ PaymentMethodDetails: { PaymentMethodDetail: CreditCard } }
```
```javascript
// XML
<PaymentMethodDetails>
    <PaymentMethodDetail>CreditCard</PaymentMethodDetail>
   <PaymentMethodDetail>SomethingElse</PaymentMethodDetail>
</PaymentMethodDetails>

// JSON
{ PaymentMethodDetails: { PaymentMethodDetail: [CreditCard, SomethingElse] } }
```

When I look at the `reports_get_report_request_list_multiple.xml` fixture, the `ReportRequestInfo` sections are not nested inside a parent element, they are listed at the root of the response, so instead of this:
```
<GetReportRequestListResult>
    <NextToken>2YgYW55IPQhcm5hbCBwbGVhc3VyZS4=</NextToken>
    <ReportRequestInfos>
        <ReportRequestInfo></ReportRequestInfo>
        <ReportRequestInfo></ReportRequestInfo>
        <ReportRequestInfo></ReportRequestInfo>
    </ReportRequestInfos>
</GetReportRequestListResult>
```
it's just this:
```
<GetReportRequestListResult>
    <NextToken>2YgYW55IPQhcm5hbCBwbGVhc3VyZS4=</NextToken>
    <ReportRequestInfo></ReportRequestInfo>
    <ReportRequestInfo></ReportRequestInfo>
    <ReportRequestInfo></ReportRequestInfo>
</GetReportRequestListResult>
```
which is really strange, but this is the reason why ensureArray doesn't work.

Reply:
Huh is this the actual response we got or is it from a doc sample @justinemmanuelmercado ?


<GetReportRequestListResult>
 <NextToken>2YgYW55IPQhcm5hbCBwbGVhc3VyZS4=</NextToken>
    <ReportRequestInfo></ReportRequestInfo>
    <ReportRequestInfo></ReportRequestInfo>
    <ReportRequestInfo></ReportRequestInfo>
</GetReportRequestListResult>

Reply:
@moltar The fixture is not an actual response, I made it before integration tests for ReportRequestInfo passed, but it's the same structure as what Amazon responds with. I'll replace it with Polly's recording

Reply:
```xml
<GetReportRequestListResponse xmlns="http://mws.amazonaws.com/doc/2009-01-01/">
  <GetReportRequestListResult>
    <HasNext>false</HasNext>
    <ReportRequestInfo>
      <ReportType>_GET_MERCHANT_LISTINGS_DATA_</ReportType>
      <ReportProcessingStatus>_DONE_</ReportProcessingStatus>
      <EndDate>2020-06-11T07:44:22+00:00</EndDate>
      <Scheduled>false</Scheduled>
      <ReportRequestId>51781018424</ReportRequestId>
      <StartedProcessingDate>2020-06-11T07:44:25+00:00</StartedProcessingDate>
      <SubmittedDate>2020-06-11T07:44:22+00:00</SubmittedDate>
      <StartDate>2020-06-11T07:44:22+00:00</StartDate>
      <CompletedDate>2020-06-11T07:44:32+00:00</CompletedDate>
      <GeneratedReportId>21175694446018424</GeneratedReportId>
    </ReportRequestInfo>
    <ReportRequestInfo>
      <ReportType>_GET_FLAT_FILE_OPEN_LISTINGS_DATA_</ReportType>
      <ReportProcessingStatus>_DONE_</ReportProcessingStatus>
      <EndDate>2020-06-11T07:41:44+00:00</EndDate>
      <Scheduled>false</Scheduled>
      <ReportRequestId>51780018424</ReportRequestId>
      <StartedProcessingDate>2020-06-11T07:41:48+00:00</StartedProcessingDate>
      <SubmittedDate>2020-06-11T07:41:44+00:00</SubmittedDate>
      <StartDate>2020-06-11T07:41:44+00:00</StartDate>
      <CompletedDate>2020-06-11T07:41:56+00:00</CompletedDate>
      <GeneratedReportId>21178489344018424</GeneratedReportId>
    </ReportRequestInfo>
    <ReportRequestInfo>
      <ReportType>_GET_FLAT_FILE_ORDERS_DATA_</ReportType>
      <ReportProcessingStatus>_DONE_NO_DATA_</ReportProcessingStatus>
      <EndDate>2020-06-11T07:40:49+00:00</EndDate>
      <Scheduled>false</Scheduled>
      <ReportRequestId>51779018424</ReportRequestId>
      <StartedProcessingDate>2020-06-11T07:40:52+00:00</StartedProcessingDate>
      <SubmittedDate>2020-06-11T07:40:49+00:00</SubmittedDate>
      <StartDate>2020-06-10T07:40:49+00:00</StartDate>
      <CompletedDate>2020-06-11T07:40:59+00:00</CompletedDate>
    </ReportRequestInfo>
    <ReportRequestInfo>
      <ReportType>_GET_V2_SETTLEMENT_REPORT_DATA_FLAT_FILE_</ReportType>
      <ReportProcessingStatus>_DONE_</ReportProcessingStatus>
      <EndDate>2020-05-22T07:52:49+00:00</EndDate>
      <Scheduled>false</Scheduled>
      <ReportRequestId>51778018404</ReportRequestId>
      <StartedProcessingDate>2020-05-22T08:16:14+00:00</StartedProcessingDate>
      <SubmittedDate>2020-05-22T08:16:14+00:00</SubmittedDate>
      <StartDate>2020-04-24T07:52:49+00:00</StartDate>
      <CompletedDate>2020-05-22T08:16:14+00:00</CompletedDate>
      <GeneratedReportId>20812705939018404</GeneratedReportId>
    </ReportRequestInfo>
    <ReportRequestInfo>
      <ReportType>_GET_V2_SETTLEMENT_REPORT_DATA_FLAT_FILE_</ReportType>
      <ReportProcessingStatus>_DONE_</ReportProcessingStatus>
      <EndDate>2020-04-24T07:52:49+00:00</EndDate>
      <Scheduled>false</Scheduled>
      <ReportRequestId>51777018376</ReportRequestId>
      <StartedProcessingDate>2020-04-24T08:10:53+00:00</StartedProcessingDate>
      <SubmittedDate>2020-04-24T08:10:53+00:00</SubmittedDate>
      <StartDate>2020-03-13T07:52:49+00:00</StartDate>
      <CompletedDate>2020-04-24T08:10:53+00:00</CompletedDate>
      <GeneratedReportId>20303555135018376</GeneratedReportId>
    </ReportRequestInfo>
    <ReportRequestInfo>
      <ReportType>_GET_V2_SETTLEMENT_REPORT_DATA_FLAT_FILE_</ReportType>
      <ReportProcessingStatus>_DONE_</ReportProcessingStatus>
      <EndDate>2020-03-13T07:52:49+00:00</EndDate>
      <Scheduled>false</Scheduled>
      <ReportRequestId>51776018334</ReportRequestId>
      <StartedProcessingDate>2020-03-13T08:21:03+00:00</StartedProcessingDate>
      <SubmittedDate>2020-03-13T08:21:03+00:00</SubmittedDate>
      <StartDate>2020-02-14T07:52:49+00:00</StartDate>
      <CompletedDate>2020-03-13T08:21:03+00:00</CompletedDate>
      <GeneratedReportId>19571121666018334</GeneratedReportId>
    </ReportRequestInfo>
  </GetReportRequestListResult>
  <ResponseMetadata>
    <RequestId>3930447a-fff6-4dc9-9ac4-9f063da318b3</RequestId>
  </ResponseMetadata>
</GetReportRequestListResponse>

```

Reply:
The above is from Polly

Reply:
@justinemmanuelmercado So does that solve the problem now?

Reply:
Oh yeah, no problem really, I found out that there was a parsing issue when I made the integration tests, I tried out `ensureArray` but it still had issues parsing the response. So I replaced it with 
```typescript
ReportRequestInfo: optional(oneOf([array(ReportRequestInfo), ReportRequestInfo, exactly('')]))
```
and it worked, but I was wondering why `ensureArray` wasn't working. But gigobyte cleared up why it didn't work. Might have to see what to do about this type of response because AFAIK `GetReportList` has a similar response structure